pkg/controllers/provisioning/v1alpha1/allocation/constraints.go

```diff
@@ -101,6 +105,32 @@ func (c *Constraints) getLabels(provisioner *v1alpha1.Provisioner, pod *v1.Pod)
 	)
 }
 
+func (c *Constraints) getZones(provisioner *v1alpha1.Provisioner, pod *v1.Pod) []string {
+	// Pod may override zone
+	if zone, ok := pod.Spec.NodeSelector[ZoneLabelKey]; ok {
+		return []string{zone}
+	}
+	// Default to provisioner constraints
+	if provisioner.Spec.Zones != nil {
+		return provisioner.Spec.Zones
+	}
+	// Otherwise unconstrained
+	return nil
+}
+
+func (c *Constraints) getInstanceTypes(provisioner *v1alpha1.Provisioner, pod *v1.Pod) []string {
+	// pod may override instance type
+	if instanceType, ok := pod.Spec.NodeSelector[InstanceTypeLabelKey]; ok {
+		return []string{instanceType}
+	}
+	// Default to provisioner constraints
+	if provisioner.Spec.Zones != nil {
```

Comment:
This should be `Spec.InstanceTypes` and `if len(provisioner.Spec.InstanceTypes) == 0`